File: spiralMatrixII.cpp
class Solution {
public:
    vector<vector<int> > generateMatrix(int n) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        vector<vector<int>> result(n, vector<int>(n));
        if(n == 0) return result;
        int start = 0, end = n - 1;
        int num = 1;
        while(start < end){
            // write top
            for(int i = start; i < end; i++){
                result[start][i] = num;
                num++;
            }
            // write right
            for(int i = start; i < end; i++){
                result[i][end] = num;
                num++;
            }
            // write bottom
            for(int i = end; i > start; i--){
                result[end][i] = num;
                num++;
            }
            // write left
            for(int i = end; i > start; i--){
                result[i][start] = num;
                num++;
            }
            start++;
            end--;
        }
        // odd # of rows/columns
        if(start == end) result[start][start] = num;
        return result;
    }
};

File: spiralMatrixII.java
public class Solution{
	public int[][] generateMatrix(int n){
		int [][] matrix = new int[n][n];
		int start = 0, end = n - 1, num = 1;
		while(start < end){
			for(int j = start; j < end; j++){
				matrix[start][j] = num;
				num++;
			}
			for(int i = start; i < end; i++){
				matrix[i][end] = num;
				num++;
			}
			for(int j = end; j > start; j--){
				matrix[end][j] = num;
				num++;
			}
			for(int i = end; i > start; i--){
				matrix[i][start] = num;
				num++;
			}
			start++;
			end--;
			
				
		}
		if(start == end)
			num[start][end] = num;
		return matrix;
	}
}
